tmt/utils.py

```diff
@@ -1166,6 +1178,20 @@ def copytree(
     return dst
 
 
+def get_full_metadata(fmf_tree_path: str, node_path: str) -> Any:
+    """
+    Get full metadata for a node in any fmf tree
+
+    Go through fmf tree nodes using given relative node path
+    and return full data as dictionary
+    """
+    node: fmf.Tree = fmf.Tree(fmf_tree_path)
+    for node_name in node_path.strip('/').split('/'):
+        if node_name:
+            node = node.children[node_name]
+    return node.data
```

Comment:
So you're looking for a specific node in the `fmf.Tree`, right? What about this:

```python
fmf.Tree(fmf_tree_path).find(node_path).data
```

You could use the `Tree.find` method to easily locate the desired node. Isn't this what you are looking for?